java/clients/src/main/java/sleeper/clients/status/report/QueryTrackerReport.java

```diff
@@ -79,7 +79,7 @@ public static void main(String[] args) {
             QueryTrackerReporter reporter = getReporter(args, 1);
             TrackerQuery queryType = optionalArgument(args, 2)
                     .map(QueryTrackerReport::readTypeArgument)
-                    .orElse(promptForQueryType());
+                    .orElseGet(QueryTrackerReport::promptForQueryType);
```

Comment:
I noticed promptForQueryType can return null, in which case this will fail. That also means that readTypeArgument can result in an empty optional even when it's prompted. It'll then prompt again, which doesn't seem like the expected behaviour.

What should happen in the case where TrackerQueryPrompt.from is currently returning null?

This is still an improvement over how it was before. You could raise a separate issue for that.

Reply:
I think in other places we just keep looping until a valid input is provided.